test/Microsoft.AspNet.Mvc.DataAnnotations.Test/DataAnnotationsModelValidatorTest.cs

```diff
@@ -284,11 +314,111 @@ public void Validate_IsValidFalse_StringLocalizerReturnsLocalizerErrorMessage()
+                    },
+                    {
+                        new CreditCardAttribute() { ErrorMessage = LocalizationKey },
+                        pattern,
+                        new object[] { nameof(SampleModel), "CreditCard" }
+                    },
+                    {
+                        new StringLengthAttribute(length) { MinimumLength = 1, ErrorMessage = LocalizationKey },
+                        "",
+                        new object[] { nameof(SampleModel), 1, length }
+                    },
+                    {
+                        new RangeAttribute(0, length) { ErrorMessage = LocalizationKey },
+                        pattern,
+                        new object[] { nameof(SampleModel), 0, length}
+                    },
+                    {
+                        new EmailAddressAttribute() { ErrorMessage = LocalizationKey },
+                        pattern,
+                        new object[] { nameof(SampleModel), "EmailAddress" }
+                    },
+                    {
+                        new PhoneAttribute() { ErrorMessage = LocalizationKey },
+                        pattern,
+                        new object[] { nameof(SampleModel), "PhoneNumber" }
+                    },
+                    {
+                        new UrlAttribute() { ErrorMessage = LocalizationKey },
+                        pattern,
+                        new object[] { nameof(SampleModel), "Url"  }
+                    }
+                };
+            }
         }
 
-        private class DerivedRequiredAttribute : RequiredAttribute
+        [Theory]
+        [MemberData(nameof(Validate_AttributesIncludeValues))]
+        public void Validate_IsValidFalse_StringLocalizerGetsArguments(
+            ValidationAttribute attribute,
+            string model,
+            object[] values)
         {
+            if (attribute is StringLengthAttribute)
+            {
+                Console.WriteLine();
+            }
+
+            // Arrange
+            var stringLocalizer = new Mock<IStringLocalizer>();
+
+            var validator = new DataAnnotationsModelValidator(
+                attribute,
+                stringLocalizer.Object,
+                validationAttributeAdapterProvider: new ValidationAttributeAdapterProvider());
+
+            var metadata = _metadataProvider.GetMetadataForType(typeof(SampleModel));
+            var validationContext = new ModelValidationContext(
+                actionContext: new ActionContext(),
+                modelMetadata: metadata,
+                metadataProvider: _metadataProvider,
+                container: null,
+                model: model);
+
+            // Act
+            validator.Validate(validationContext);
+
+            // Assert
+            var json = Newtonsoft.Json.JsonConvert.SerializeObject(values) + " " + attribute.GetType().Name;
+
+            stringLocalizer.Verify(l => l[LocalizationKey, values], json);
```

Comment:
If we don't assign an ErrorMessage to the attribute then it's null and localization won't happen, as per https://github.com/aspnet/Mvc/pull/3649/files#diff-d057e6d613cd8f4ca6565fb70c62c5a3R93